src/libraries/Hooks.sol

```diff
@@ -259,27 +272,30 @@ library Hooks {
         IPoolManager.SwapParams memory params,
         BalanceDelta swapDelta,
         bytes calldata hookData,
-        int128 hookDeltaSpecified
-    ) internal returns (BalanceDelta swapperDelta, BalanceDelta hookDelta) {
+        BeforeSwapDelta beforeSwapHookReturn
+    ) internal returns (BalanceDelta, BalanceDelta) {
         if (msg.sender == address(self)) return (swapDelta, BalanceDeltaLibrary.ZERO_DELTA);
 
-        int128 hookDeltaUnspecified;
-        swapperDelta = swapDelta;
+        int128 hookDeltaSpecified = beforeSwapHookReturn.getSpecifiedDelta();
+        int128 hookDeltaUnspecified = beforeSwapHookReturn.getUnspecifiedDelta();
+
         if (self.hasPermission(AFTER_SWAP_FLAG)) {
-            hookDeltaUnspecified = self.callHookWithReturnDelta(
+            hookDeltaUnspecified += self.callHookWithReturnDelta(
                 abi.encodeWithSelector(IHooks.afterSwap.selector, msg.sender, key, params, swapDelta, hookData),
                 self.hasPermission(AFTER_SWAP_RETURNS_DELTA_FLAG)
             ).toInt128();
         }
 
+        BalanceDelta hookDelta;
         if (hookDeltaUnspecified != 0 || hookDeltaSpecified != 0) {
```

Comment:
Just making sure that in the case hookDeltaUnspecified is nonzero and then on line 283 they return something to make it 0 again thats OK. Ie we don't actually use hookDeltaUnspecified in any way before or during the internal swap call?

Reply:
yeah nice call out, but hookDeltaUnspecified is only used after the afterswap hook so if they cancel out that was just a waste of gas